### PR TITLE
Jd/service duals fix

### LINKED ISSUE #885: Dual variables not available for service related constraints
**Describe the bug**
Dual variables are not available for service related constraints in the results, even though specified in the template.

### PR DESCRIPTION
**Pull request by jd-lara**
closes #885

### CODE REVIEW

File: src/devices_models/devices/common/add_constraint_dual.jl [hunk 1]
```diff
@@ -44,15 +44,40 @@ function add_constraint_dual!(
...
 end
 
 function assign_dual_variable!(
     container::OptimizationContainer,
     constraint_type::Type{<:ConstraintType},
     devices::U,
-    formulation,
-) where {U <: Union{Vector{D}, IS.FlattenIteratorWrapper{D}}} where {D <: PSY.Component}
+    ::AbstractDeviceFormulation,
+) where {U <: Union{Vector{D}, IS.FlattenIteratorWrapper{D}}} where {D <: PSY.Device}
```
Comment (Nongchao) [inline src/devices_models/devices/common/add_constraint_dual.jl:80]: Should this still be `Component`? The error I got seems to indicate it's trying to apply this function to `buses`?

Comment (jd-lara) [reply src/devices_models/devices/common/add_constraint_dual.jl:80]: Not all components have device formulations. In my last commit I addressed this.

File: src/devices_models/devices/common/add_constraint_dual.jl [hunk 2]
```diff
@@ -57,7 +57,7 @@ function assign_dual_variable!(
     container::OptimizationContainer,
     constraint_type::Type{<:ConstraintType},
     service::D,
-    ::AbstractServiceFormulation,
+    ::Type{AbstractServiceFormulation},
```
Comment (Nongchao) [inline src/devices_models/devices/common/add_constraint_dual.jl:60]: @jd-lara Should this be `::Type{<:AbstractServiceFormulation}`?

Comment (jd-lara) [reply src/devices_models/devices/common/add_constraint_dual.jl:60]: Yes

**Pull request merged**